src/test/java/com/commercetools/sync/products/helpers/VariantReferenceResolverTest.java

```diff
@@ -26,461 +24,418 @@
-        resolvedReference.put(REFERENCE_ID_FIELD, PRODUCT_ID);
-        assertThat(resolvedSet).containsExactly(resolvedReference);
+        final List<AssetDraft> resolvedBuilderAssets = resolvedBuilder.getAssets();
+        assertThat(resolvedBuilderAssets).isEmpty();
     }
 
-    @Disabled("Fails due to possible bug on https://github.com/FasterXML/jackson-databind/issues/2442")
     @Test
-    void resolveAttributeReference_WithNullReferenceInSetAttribute_ShouldResolveReferences() {
-        final ObjectNode productReference = getProductReferenceWithRandomId();
-        final AttributeDraft productReferenceAttribute =
-            getProductReferenceSetAttributeDraft("foo", productReference, null);
+    void resolveAssetsReferences_WithAssetReferences_ShouldResolveAssets() {
+        final CustomFieldsDraft customFieldsDraft = CustomFieldsDraft
+            .ofTypeIdAndJson("customTypeId", new HashMap<>());
 
-        final AttributeDraft resolvedAttributeDraft =
-            referenceResolver.resolveAttributeReference(productReferenceAttribute)
-                             .toCompletableFuture().join();
-        assertThat(resolvedAttributeDraft).isNotNull();
-        assertThat(resolvedAttributeDraft.getValue()).isNotNull();
+        final AssetDraft assetDraft = AssetDraftBuilder.of(emptyList(), ofEnglish("assetName"))
+                                                       .custom(customFieldsDraft)
+                                                       .build();
 
-        final Spliterator<JsonNode> attributeReferencesIterator = resolvedAttributeDraft.getValue().spliterator();
-        assertThat(attributeReferencesIterator).isNotNull();
-        final Set<JsonNode> resolvedSet = StreamSupport.stream(attributeReferencesIterator, false)
-                                                       .collect(Collectors.toSet());
-        assertThat(resolvedSet).isNotEmpty();
-        final ObjectNode resolvedReference = JsonNodeFactory.instance.objectNode();
-        resolvedReference.put(REFERENCE_TYPE_ID_FIELD, "product");
-        resolvedReference.put(REFERENCE_ID_FIELD, PRODUCT_ID);
-        assertThat(resolvedSet).containsExactly(resolvedReference);
+        final ProductVariantDraftBuilder productVariantDraftBuilder =
+            ProductVariantDraftBuilder.of().assets(singletonList(assetDraft));
+
+        final ProductVariantDraftBuilder resolvedBuilder = referenceResolver
+            .resolveAssetsReferences(productVariantDraftBuilder).toCompletableFuture().join();
+
+        final List<AssetDraft> resolvedBuilderAssets = resolvedBuilder.getAssets();
+        assertThat(resolvedBuilderAssets).isNotEmpty();
+        final AssetDraft resolvedAssetDraft = resolvedBuilderAssets.get(0);
+        assertThat(resolvedAssetDraft).isNotNull();
+        assertThat(resolvedAssetDraft.getCustom()).isNotNull();
+        assertThat(resolvedAssetDraft.getCustom().getType().getId()).isEqualTo("typeId");
```

Comment:
maybe use constant for typeId ?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/442/commits/b936197ff501e01516f748cf9335e18f59ec1aa3